### DISCUSSION TITLE
BTTV Pro Sub Badges

### DISCUSSION THREAD

**Discussion by PepegePaul** [Chatterino/chatterino2#4323] [Ideas]
So theres this new badge for BTTV pro subscribers, might be cool to implement in Chatterino if you want to of course
![example of badge in webchat](https://i.imgur.com/lOjdpFi.png)

**Comment by Felanbird** [comment]
Relevant commit: https://github.com/night/betterttv/commit/89860b41bb6878ca929da4ba50287fb7ddf65299 no idea where the list is though

**Reply by pajlada** [reply]
I'm happy to add support for BTTV badges if the API is simple (e.g. like FFZ's API)

**Reply by LosFarmosCTL** [reply]
Badges appear to be handled per-user through their WebSocket, exactly how I have no idea and since it's undocumented whoever implements it would have to do some more digging around in the extensions code. So probably not really all that simple, might be OOS to add another websocket connection just for having some badges (and no idea if it even scales from a single webchat instance to a chatterino with 100+ channels).

I remember something about the personal emotes not being implemented due to that issue and since, from what I can tell, it's the same websocket, this might be the case here as well.

**Reply by Felanbird** [reply]
> I remember something about the personal emotes not being implemented due to that issue and since, from what I can tell, it's the same websocket, this might be the case here as well.

We actually do consume this websocket now for auto-updating-bttv-emotes - but it does still seem like it would be a hassle compared to the other services, including our own, that give us a nice JSON response.

**Reply by ZonianMidian** [reply]
> I'm happy to add support for BTTV badges if the API is simple (e.g. like FFZ's API)

The documentation for its API has recently been released: https://betterttv.com/developers/api#badges

**Reply by LosFarmosCTL** [reply]
Oh very nice, that API looks like it can easily be used for chatterino.

**Reply by PepegePaul** [reply]
Thank you guys :)

**Reply by ZonianMidian** [reply]
> The documentation for its API has recently been released: https://betterttv.com/developers/api#badges

As @pajlada commented to me in his chat, the API provides all badges except for Pro user badges (the relevant ones in this discussion). So consulting, the correct implementation of these is via WebSocket (as @LosFarmosCTL mentioned above).
Documentation: https://betterttv.com/developers/websocket#user-update

![image](https://user-images.githubusercontent.com/57513632/214979661-b2121b85-2fbf-4602-90bf-82a59b0722fb.png)

**Reply by Felanbird** [reply]
Tell him we don't want that

**Reply by LosFarmosCTL** [reply]
BTTV really trying to be as good and useful as the Twitch APIs themselves huh

**Reply by ZonianMidian** [reply]
> BTTV really trying to be as good and useful as the Twitch APIs themselves huh

I honestly thought that in this version of the API being released at the same time as the Pro user badges would be implemented in the endpoint, a real shame that this is not the case, it makes it useless...

**Reply by LosFarmosCTL** [reply]
It's most likely down to him not wanting to share how many subscribers there are (and therefore the revenue of the company), which ig is a valid reason to not provide that information, but it is crippling what 3rd parties can do with it.

**Reply by pajlada** [reply]
Perfectly fair for pro badges to be websocket only, if someone is interested I'm still happy to implement their developer badge rest API in Chatterino.
